### PR TITLE
multistep macos setup

### PR DESCRIPTION
**Pull request by garyverhaegen-da**
This updates the macOS node setup isntructions to avoid repeating identical work and network traffic across all machines through initialization by building a "daily" image with all the tools and code we need.

CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

**Comment by garyverhaegen-da** [comment]
I have not been able to fully test this as my local hard drive is not big enough to contain the macinbox image, the step 2 VM and the resulting box, but I am fairly confident it should work. 🤞

File: infra/macos/2-common-box/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+You can refer to the instructions in [this README](../1-create-box/README.md),
+or, for a more automated setup, use the following set of commands (which will
+require a sudoer password at some points):
+
+```
+/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)" < /dev/null
+brew cask install virtualbox
+brew cask install vagrant
+cd $(mktemp -d)
+wget https://download.virtualbox.org/virtualbox/6.1.4/Oracle_VM_VirtualBox_Extension_Pack-6.1.4.vbox-extpack
```
Comment (nycnewman) [inline infra/macos/2-common-box/README.md:21]: Any reason not to move to 6.1.6? What does cask install by default?

Comment (garyverhaegen-da) [reply infra/macos/2-common-box/README.md:21]: Oh I just did not touch that part. I'll update.

**Comment by garyverhaegen-da** [comment]
@nycnewman I've updated to be compatible with your changes so far. With the new settings for VB, I've also been able to test locally: the last step of `2-common-box` failed on writing to disk when initializing Bazel (but had by that point successfully installed the XCode CLI tools, homebrew, and nix), and because that step is allowed to fail I was also able to test step 3, i.e. build on top of the produced box. I was then able to connect to Azure and run a simple job. (Connected to my own Azure org and the job was "`exit 0`".)

**Comment by garyverhaegen-da** [comment]
I've (finally) finished testing this to the point where I'm confident it should work as advertized. I can confirm the only thing the "other" machines need is the `.box` file, and the `vagrant box add` command copies it over, so there is no need to keep the `.box` file after running that either.

50GB seems to be just enough to complete one build of the daml repo and reach a point where the cache is fully initialized, so I set that as the "recommended" settings for the VBox version.

File: infra/macos/scripts/step2.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+cd ~/daml/infra/macos/2-common-box
+
+GUEST_NAME=$(HOSTNAME) vagrant up
+
+vagrant package --output ~/images/initialized-$(date +%Y%m%d).box
+
+vagrant destroy -f
+
+cd ~
+./copyfile.sh images/initialized-$(date +%Y%m%d).box
```
Comment (garyverhaegen-da) [inline infra/macos/scripts/step2.sh:14]: Should `copyfile.sh` be checked in too?

**Comment by garyverhaegen-da** [comment]
@nycnewman Thanks for looking at this over the weekend! I have added copyright notices to the step{1,2}.sh files (otherwise CI will refuse to merge).

My understanding is we now have one node (builder1gpjj7) using the new scheme, and will keep that for a few days to observe its behaviour. Assuming everything goes well, please merge this PR once the rest of the cluster is switched over.

**Comment by garyverhaegen-da** [comment]
Confirmed with @nycnewman that this has now been fully deployed.

**Pull request merged**